plugins/woocommerce/client/blocks/docs/internal-developers/block-client-apis/checkout/checkout-api.md

```diff
@@ -65,13 +65,24 @@ The following actions can be dispatched from the Checkout data store:
 -   `__internalSrocessCheckoutResponse( response: CheckoutResponse )`: This is a thunk that will extract the paymentResult from the CheckoutResponse, and dispatch 3 actions: `__internalSetRedirectUrl`, `__internalSetPaymentResult` and `__internalSetAfterProcessing`.
 -   `__internalSetRedirectUrl( url: string )`: Set `state.redirectUrl` to `url`
 -   `__internalSetHasError( trueOrFalse: bool )`: Set `state.hasError` to `trueOrFalse`
--   `__internalIncrementCalculating()`: Increment `state.calculatingCount`
--   `__internalDecrementCalculating()`: Decrement `state.calculatingCount`
+-   `__internalStartCalculation`: Increment `state.calculatingCount`
+-   `__internalFinishCalculation`: Decrement `state.calculatingCount`
 -   `__internalSetCustomerId( id: number )`: Set `state.customerId` to `id`
 -   `__internalSetUseShippingAsBilling( useShippingAsBilling: boolean )`: Set `state.useShippingAsBilling` to `useShippingAsBilling`
 -   `__internalSetShouldCreateAccount( shouldCreateAccount: boolean )`: Set `state.shouldCreateAccount` to `shouldCreateAccount`
 -   `__internalSetOrderNotes( orderNotes: string )`: Set `state.orderNotes` to `orderNotes`
 -   `setExtensionData( namespace: string, extensionData: Record< string, unknown > )`: Set `state.extensionData` to `extensionData`
+-   `disableCheckoutFor( asyncFunc: () => Promise< unknown > )`: A thunk that allows to disable checkout button until promise returned by `asyncFunc` gets resolved. Relies on the same internal state as the one powering the tracking of pending calculations.
```

Comment:
> Relies on the same internal state as the one powering the tracking of pending calculations.

Not sure if we should mention this, as we're going to remove the state for pending calculations in the future.

Reply:
I didn't mention it in the developer docs (top level `/docs` directory), but since these are more targeted at internal developers and discuss internals in detail, then I figured it is a worthy addition. If we decide to remove the state tracking pending calculations these docs will need an overhaul.

If you have other suggestions, I am happy to apply them.